weather.go

```diff
@@ -1,5 +1,138 @@
 package weather
 
-func Get(location, token string) ([]byte, error) {
-	return nil, nil
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math"
+	"net/http"
+	"os"
+	"strings"
+)
+
+type apiResponse struct {
+	Name    string
+	Weather []struct {
+		Main string
+	}
+	Main struct {
+		Temp float64
+	}
+}
+
+type apiErrorResponse struct {
+	Cod     string
+	Message string
+}
+
+type Conditions struct {
+	City               string
+	OneWord            string
+	TemperatureCelsius float64
+}
+
+func RunCLI(args []string) (output string) {
+
+	token := os.Getenv("WEATHERAPP_TOKEN")
+	if len(token) == 0 {
+		fmt.Fprintf(os.Stderr, "please set env variable, WEATHERAPP_TOKEN \n")
+		os.Exit(2)
+	}
+
+	request, err := Request(args, token)
```

Comment:
I'm not sure there's value in returning a `*http.Request` here rather than simply the URL as a string. A string is easier to test, isn't it? We're essentially making an unnecessary call to `http.NewRequest` only in order to return this type.